serde_arrow/src/arrow_impl/api.rs

```diff
@@ -76,17 +73,14 @@ impl ArrowBuilder {
     /// given fields.
     ///
     pub fn new(fields: &[Field]) -> Result<Self> {
-        let fields = fields
-            .iter()
-            .map(GenericField::try_from)
-            .collect::<Result<Vec<_>>>()?;
-        Ok(Self(generic::GenericBuilder::new_for_arrays(&fields)?))
+        let schema = SerdeArrowSchema::from_arrow_fields(fields)?;
+        Ok(Self(ArrayBuilder::new(&schema)?))
     }
 
     /// Add a single record to the arrays
     ///
     pub fn push<T: Serialize + ?Sized>(&mut self, item: &T) -> Result<()> {
-        self.0.push(item)
+        self.0.extend(&[item])
```

Comment:
The update of the validity buffer is performed for each item (it has to be, as each item can be None or Some). However, the outer `ListBuilder` is always created [with a None nullability bitmap](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/internal/serialization_ng/array_builder.rs#L100). Hence no updates of the bit map will be performed. 

While in principle, it could make sense to build a specialized builder for this occasion without a nullability bit map at all, my hope was that performance overead is minimal.

Reply:
Oh. Actually good catch. The reason this works is: the outer `ListBuilder` is just used for convinience. When the arrays are constructed only inner struct builder is acutally used (see [here](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/arrow_impl/serialization.rs#L18) and [here](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/internal/serialization_ng/array_builder.rs#L275)) Maybe it makes sense to add a dedicated builder for the outer list, without all the additional baggage (most notably the offset array).

Regarding the nullability itself: the outer list builder is [always constructed with a None validity](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/internal/serialization_ng/array_builder.rs#L100), hence any nullability updates will be [no-ops](https://github.com/chmp/serde_arrow/blob/e48662dd5a87a98de8d4264215bc24ce8f71af7d/serde_arrow/src/internal/serialization_ng/utils.rs#L23).